### PR TITLE
Android: Add support to upload files after recording audio / video

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206248386427112/f

### Description
Added support to capture video.

### Steps to test this PR

_Capture enabled with "video" only_
- [x] Install from this branch
- [x] Open https://jsbin.com/gicinimaca/1
- [x] Tap on Browse 
- [x] Go through the permissions flow
- [x] See the camera opens to record a video
- [x] Record a video and notice the filename has the `.mp4` extension

_Capture enabled with "audio" only_
- [x] Open https://jsbin.com/zutifupuxu/1
- [x] Tap on Browse
- [x] Go through the permissions flow
- [x] See the camera opens to record audio if an app is installed on your device. If the result is unexpected (you're prompted to choose a file) on your device, check if it behaves the same in Chrome.
- [x] Record audio and notice the filename has the `.m4a` extension

_Capture enabled with "audio" and other types_
- [x] Open https://jsbin.com/wuhasadowe/1
- [x] Tap on Browse
- [x] You're prompted to choose the content from device

### CODE REVIEW

**Comment by anikiki** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4060?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

<b>In this stack:</b> <code>📱 Android: Enhancing video capture capabilities</code>
* `develop`
  * **#4052** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4052?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#4060** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4060?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @anikiki and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 1]
```diff
@@ -744,16 +745,16 @@ class BrowserTabFragment :
...
                 is ImageCaptured -> pendingUploadTask?.onReceiveValue(arrayOf(Uri.fromFile(it.file)))
-                CouldNotCapturePermissionDenied -> {
+                is CouldNotCapturePermissionDenied -> {
                     pendingUploadTask?.onReceiveValue(null)
                     activity?.let { activity ->
-                        externalCameraLauncher.showPermissionRationaleDialog(activity)
+                        externalCameraLauncher.showPermissionRationaleDialog(activity, it.input)
                     }
                 }
                 NoImageCaptured -> pendingUploadTask?.onReceiveValue(null)
```
Comment (marcosholgado) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:754]: Should this also be `is NoImageCaptured` ?

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt [hunk 2]
```diff
@@ -1892,19 +1903,29 @@ class BrowserTabViewModel @Inject constructor(
...
+        val cameraHardwareAvailable = cameraHardwareChecker.hasCameraHardware()
 
         command.value = when {
-            fileChooserParams.isCaptureEnabled && acceptsOnlyImage && cameraHardwareAvailable ->
-                ShowImageCamera(filePathCallback)
-            (fileChooserParams.acceptTypes.any { it.startsWith("image/") } && cameraHardwareAvailable) ->
-                ShowExistingImageOrCameraChooser(filePathCallback, fileChooserRequestedParams)
-            else -> ShowFileChooser(filePathCallback, fileChooserRequestedParams)
+            fileChooserParams.isCaptureEnabled -> {
+                when {
```
Comment (marcosholgado) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt:1912]: Could we have several acceptTypes or it's either image, video, audio or others? i.e. can we have audio AND video?

**Review by marcosholgado — APPROVED** [review]
LGTM!

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt (hunk 2 above)
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt:1912]: When `capture` is enabled, it has to be used with one of the types. However, this is not enforced. If the website is not correctly using this, we default to choosing a file from device. I noticed Chrome is showing all possible options. We don't have a UI at the moment that could do this, but it's a good point to address in the future.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt (hunk 1 above)
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt]: Updated, thanks :+1:

**Pull request merged**